packages/immutable-arraybuffer/src/immutable-arraybuffer-pony.js

```diff
@@ -0,0 +1,241 @@
+/* global globalThis */
+
+const {
+  ArrayBuffer,
+  Object,
+  Reflect,
+  TypeError,
+  Uint8Array,
+  // Capture structuredClone before it can be scuttled.
+  structuredClone: optStructuredClone,
+} = globalThis;
+
+const { defineProperty, getPrototypeOf } = Object;
+const { apply, ownKeys } = Reflect;
+
+const { prototype: arrayBufferPrototype } = ArrayBuffer;
+const { slice, transfer: optTransfer } = arrayBufferPrototype;
+// @ts-expect-error TS doesn't know it'll be there
+const { get: arrayBufferByteLength } = Object.getOwnPropertyDescriptor(
```

Comment:
I don’t expect there to be a meaningful difference in behavior, but you might capture `getOwnPropertyDescriptor` to be consistent with `defineProperty` and `getPrototypeOf`.